addons/lunch/static/src/js/lunch.js

```diff
@@ -0,0 +1,63 @@
+(function() {
+
+    "use strict";
+
+    var QWeb = openerp.web.qweb,
+        lunch = openerp.lunch = {};
+
+    lunch.previousorder = openerp.web.form.AbstractField.extend(openerp.web.form.ReinitializeWidgetMixin,{
+        template : 'lunch.PreviousOrder',
+        update_lines: function (pref_id, records) {
+            var self = this;
+            new openerp.web.Model("lunch.order.line").call('read', [parseInt(pref_id), ['product_id', 'supplier', 'note', 'price']])
+                .then(function(res){
+                    if (res.id == parseInt(pref_id)) {
+                        if(res.note == 'false') { res.note = ''; }
+                        records.push([0, 0, {'product_id': res.product_id[0], 'note': res.note, 'supplier': res.supplier[0], 'price': res.price}]);
+                        self.field_manager.set_values({order_line_ids: records});
+                    }
+            });
+        },
```

Comment:
we can avoid this rpc call by storing data of get_previous_lunch_order_details in widget.